### PR TITLE
fix/stt_tts_selection

### PR DESCRIPTION
**Pull request by JarbasAl**
pr #22 should not have been merged, it only updated UI but not the config saving

this makes the plugin configs actually be written to file

### CODE REVIEW

File: ui/STTListMenu.qml [hunk 1]
```diff
@@ -151,7 +151,7 @@ Item {
 
                     onClicked: {
                         Mycroft.SoundEffects.playClickedSound(Qt.resolvedUrl("sounds/clicked.wav"))
-                        triggerGuiEvent("mycroft.device.confirm.stt", {"engine": model.engine, "display_name": model.display_name, "offline": model.offline})
+                        triggerGuiEvent("mycroft.device.confirm.stt", model)
```
Comment (NeonDaniel) [inline ui/STTListMenu.qml:154]: This is a breaking change; handler method in skill receives empty dict data

**Review by NeonDaniel — CHANGES_REQUESTED** [review]

File: ui/STTListMenu.qml (hunk 1 above)
Comment (JarbasAl) [reply ui/STTListMenu.qml:154]: i thought the full model would auto expand, seems we need to manually add all keys here then to ensure the final hash matches, this was not intended to be breaking, rather to add extra keys and ensure none are missing for hash step

Comment (JarbasAl) [reply ui/STTListMenu.qml:154]: this needs to exactly match
```python
{"plugin_name": plugin_display_name,
"display_name": config.get("display_name", " "),
"offline": config.get("offline", False),
"lang": lang,
"engine": engine}
```

File: ui/TTSListMenu.qml [hunk 2]
```diff
@@ -151,7 +151,7 @@ Item {
 
                     onClicked: {
                         Mycroft.SoundEffects.playClickedSound(Qt.resolvedUrl("sounds/clicked.wav"))
-                        triggerGuiEvent("mycroft.device.confirm.tts", {"engine": model.engine, "display_name": model.display_name, "gender": model.gender, "offline": model.offline})
```
Comment (JarbasAl) [inline ui/TTSListMenu.qml:154]: this needs to exactly match
```python
{"plugin_name": plugin_display_name,
"display_name": config.get("display_name", " "),
"offline": config.get("offline", False),
"lang": lang,
"gender": config.get("gender", " "),
"engine": engine}
```

**Comment by NeonDaniel** [comment]
Appears to properly update config now (though it includes the plugin config used for the GUI which is unused and potentially conflicts with expected plugin config)

**Pull request merged**